google/resource_kms_crypto_key_test.go

```diff
@@ -0,0 +1,381 @@
+			ExpectedError: true,
+		},
+		"id is in location/keyRingName/CryptoKeyName format": {
+			ImportId:            "us-central1/test-key-ring/test-key-name",
+			ExpectedError:       false,
+			ExpectedTerraformId: "test-project/us-central1/test-key-ring/test-key-name",
+			ExpectedCryptoKeyId: "projects/test-project/locations/us-central1/keyRings/test-key-ring/cryptoKeys/test-key-name",
+			Config:              &Config{Project: "test-project"},
+		},
+		"id is in location/keyRingName/CryptoKeyName format without project in config": {
+			ImportId:      "us-central1/test-key-ring/test-key-name",
+			ExpectedError: true,
+			Config:        &Config{Project: ""},
+		},
+	}
+
+	for tn, tc := range cases {
+		cryptoKeyId, err := parseKmsCryptoKeyId(tc.ImportId, tc.Config)
+
+		if tc.ExpectedError && err == nil {
+			t.Fatalf("bad: %s, expected an error", tn)
+		}
+
+		if err != nil {
+			if tc.ExpectedError {
+				continue
+			}
+			t.Fatalf("bad: %s, err: %#v", tn, err)
+		}
+
+		if cryptoKeyId.terraformId() != tc.ExpectedTerraformId {
+			t.Fatalf("bad: %s, expected Terraform ID to be `%s` but is `%s`", tn, tc.ExpectedTerraformId, cryptoKeyId.terraformId())
+		}
+
+		if cryptoKeyId.cryptoKeyId() != tc.ExpectedCryptoKeyId {
+			t.Fatalf("bad: %s, expected CryptoKey ID to be `%s` but is `%s`", tn, tc.ExpectedCryptoKeyId, cryptoKeyId.cryptoKeyId())
+		}
+	}
+}
+
+func TestCryptoKeyNextRotationCalculation(t *testing.T) {
+	now := time.Now().UTC()
+	period, _ := time.ParseDuration("1000000s")
+
+	expected := now.Add(period).Format(time.RFC3339Nano)
+
+	timestamp, err := kmsCryptoKeyNextRotation(now, "1000000s")
+
+	if err != nil {
+		t.Fatalf("unexpected failure parsing time %s and duration 1000s: %s", now, err.Error())
+	}
+
+	if expected != timestamp {
+		t.Fatalf("expected %s to equal %s", timestamp, expected)
+	}
+}
+
+func TestCryptoKeyNextRotationCalculation_validation(t *testing.T) {
+	now := time.Now().UTC()
+
+	_, err := kmsCryptoKeyNextRotation(now, "86399s")
+
+	if err == nil {
+		t.Fatalf("Periods of less than a day should be invalid")
+	}
+
+	_, err = kmsCryptoKeyNextRotation(now, "100000.0000000001s")
+
+	if err == nil {
+		t.Fatalf("Numbers with more than 9 fractional digits are invalid")
+	}
+}
+
+func TestAccGoogleKmsCryptoKey_basic(t *testing.T) {
+	skipIfEnvNotSet(t,
+		[]string{
+			"GOOGLE_ORG",
+			"GOOGLE_BILLING_ACCOUNT",
+		}...,
+	)
+
+	projectId := "terraform-" + acctest.RandString(10)
+	projectOrg := os.Getenv("GOOGLE_ORG")
+	location := os.Getenv("GOOGLE_REGION")
+	projectBillingAccount := os.Getenv("GOOGLE_BILLING_ACCOUNT")
+	keyRingName := fmt.Sprintf("tf-test-%s", acctest.RandString(10))
+	cryptoKeyName := fmt.Sprintf("tf-test-%s", acctest.RandString(10))
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckGoogleKmsCryptoKeyWasRemovedFromState("google_kms_crypto_key.crypto_key"),
```

Comment:
Since your second `TestStep` calls this already, this is just duplicating that check- not wrong, but also not necessary.